src/Workspaces/Core/Portable/FindSymbols/Declarations/DeclarationFinder_AllDeclarations.cs

```diff
@@ -34,12 +34,23 @@ internal static partial class DeclarationFinder
                 return ImmutableArray<SymbolAndProjectId>.Empty;
             }
 
-            var (succeeded, results) = await TryFindAllDeclarationsWithNormalQueryInRemoteProcessAsync(
-                project, query, criteria, cancellationToken).ConfigureAwait(false);
-
-            if (succeeded)
+            var client = await RemoteHostClient.TryGetClientAsync(project, cancellationToken).ConfigureAwait(false);
+            if (client != null)
             {
-                return results;
+                var solution = project.Solution;
+
+                var result = await client.TryRunRemoteAsync<IList<SerializableSymbolAndProjectId>>(
+                    WellKnownServiceHubServices.CodeAnalysisService,
+                    nameof(IRemoteSymbolFinder.FindAllDeclarationsWithNormalQueryAsync),
+                    solution,
+                    new object[] { project.Id, query.Name, query.Kind, criteria },
+                    callbackTarget: null,
+                    cancellationToken).ConfigureAwait(false);
+
+                if (result.HasValue)
+                {
+                    return await RehydrateAsync(solution, result.Value, cancellationToken).ConfigureAwait(false);
+                }
```

Comment:
note: this pattern has occurred so many times.  any desire to unify it behind a helper.  i.e. 

`OOPHelpers.RunRemoteOrLocal(() => /* code to run remotely */, () => /* code to run if remote doesn't work */)`.?

Reply:
I'm working on a proposal to unify the code paths as a separate change.

Reply:
thanks!

Reply:
the helper is useless imo.

Reply:
nice!